torch/_functorch/_aot_autograd/utils.py

```diff
@@ -234,57 +237,151 @@ def maybe_to_fresh_input(idx, t, meta):
     return t
 
 
-def unlift_tokens(fw_module, fw_metadata):
+def is_with_effects(node):
+    return node.op == "call_function" and node.target.__name__ == "with_effects"
```

Comment:
nit: instead of string comparison, you can do `node.target is torch.ops.higher_order.with_effects`